conn.go

```diff
@@ -189,6 +190,12 @@ type Dialer struct {
 	// database index set, this takes precedence.
 	SelectDB string
 
+	// Protocol can be used to automatically set the RESP protocol version.
+	//
+	// If Protocol is not 0 the Dialer will send a HELLO command with the
+	// value of Protocol as version, otherwise no HELLO command will be send.
+	Protocol int
```

Comment:
This may be too much flexibility but part of me feels like this would be better as a string, in case there's some shenanigans with like a `3.5` or `4-beta` protocol in the future (or something like that). Also radix might not be connecting to redis, but to some other server that looks like redis (eg keydb), and might do its own weirdness for the protocol. wdyt?

Reply:
Makes sense. Done.